lib/webworker/WebWorkerMainTemplatePlugin.js

```diff
@@ -96,7 +101,19 @@ class WebWorkerMainTemplatePlugin {
 									},
 									contentHashType: "javascript"
 								}) +
-								");"
+								(asyncChunkLoading ? ")" : ");"),
+							...(!asyncChunkLoading
+								? []
+								: [
+										Template.indent([
+											".then(function(resp) {",
```

Comment:
Fetch doesn't throw for non OK/200 responses: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch
We must assert that error code is 200.
Not sure how about redirects though.

Reply:
Good point, please note that I was asked to implement this feature for webpack 5 first - #10361 I will add those edge cases to the TODO part of the aforementioned PR